DICOMPlugins/DICOMTID1500Plugin.py

```diff
@@ -332,6 +360,80 @@ def enumerateDuplicateNames(self, items):
+            measurement = {}
+            for measurementItem in contentItem.ContentSequence:
+              if self.isConcept(measurementItem, "trackingIdentifier"):
+                measurement['trackingIdentifier'] = measurementItem.TextValue
+              if self.isConcept(measurementItem, "trackingUniqueIdentifier"):
+                measurement['trackingUniqueIdentifier'] = measurementItem.UID
+              if self.isConcept(measurementItem, "findingSite"):
+                measurement['findingSite'] = measurementItem.ConceptCodeSequence[0].CodeMeaning
+              if self.isConcept(measurementItem, "length"):
+                for lengthItem in measurementItem.ContentSequence:
+                  measurement['polyline'] = lengthItem.GraphicData
+                  for selectionItem in lengthItem.ContentSequence:
+                    if selectionItem.RelationshipType == "SELECTED FROM":
+                      for reference in selectionItem.ReferencedSOPSequence:
+                        measurement['referencedSOPInstanceUID'] = reference.ReferencedSOPInstanceUID
+                        if hasattr(reference, "ReferencedFrameNumber") and reference.ReferencedFrameNumber != "1":
+                          print('Error - only single frame references supported')
+            measurements.append(measurement)
+
+    for measurement in contents['measurements']:
+      rulerNode = slicer.vtkMRMLAnnotationRulerNode()
+      rulerNode.SetLocked(True)
+      rulerNode.SetName(contents['personObserver'])
+
+      referenceFilePath = slicer.dicomDatabase.fileForInstance(measurement['referencedSOPInstanceUID'])
+      reference = dicom.read_file(referenceFilePath)
+      origin = numpy.array(reference.ImagePositionPatient)
+      alongColumnVector = numpy.array(reference.ImageOrientationPatient[:3])
+      alongRowVector = numpy.array(reference.ImageOrientationPatient[3:])
+      alongColumnVector *= reference.PixelSpacing[1]
+      alongRowVector *= reference.PixelSpacing[0]
+      col1,row1,col2,row2 = measurement['polyline']
+      lpsToRAS = numpy.array([-1,-1,1])
+      p1 = (origin + col1 * alongColumnVector + row1 * alongRowVector) * lpsToRAS
+      p2 = (origin + col2 * alongColumnVector + row2 * alongRowVector) * lpsToRAS
+      rulerNode.SetPosition1(*p1)
+      rulerNode.SetPosition2(*p2)
+
+      rulerNode.Initialize(slicer.mrmlScene)
+      colorIndex = 1 + len(slicer.util.getNodesByClass('vtkMRMLAnnotationRulerNode'))
+      colorNode = slicer.util.getNode("GenericAnatomyColors")
+      color = [0,]*4
+      colorNode.GetColor(colorIndex, color)
+      rulerNode.GetDisplayNode().SetColor(*color[:3])
+      rulerNode.GetAnnotationTextDisplayNode().SetColor(*color[:3])
+      rulerNode.GetAnnotationPointDisplayNode().SetColor(*color[:3])
+      rulerNode.GetAnnotationPointDisplayNode().SetGlyphScale(2)
+      rulerNode.GetAnnotationLineDisplayNode().SetLabelPosition(random.random())
```

Comment:
I remember in the past we had a pointer from fiducials to the SOPInstanceUIDs when associated with DICOM. Can something like this be added here as well?

Reply:
Yes, that would make sense at some point.  We should visit that when we add export because I really think we want round-trip support.  This really needs to be migrated to the Markups anyway.

Reply:
created issue https://github.com/QIICR/QuantitativeReporting/issues/249